src/formatting/imports.rs

```diff
@@ -673,18 +703,48 @@ fn merge_rest(
 }
 
 fn merge_use_trees_inner(trees: &mut Vec<UseTree>, use_tree: UseTree, merge_by: SharedPrefix) {
-    let similar_trees = trees
-        .iter_mut()
-        .filter(|tree| tree.share_prefix(&use_tree, merge_by));
+    struct SimilarTree<'a> {
+        similarity: usize,
+        path_len: usize,
+        tree: &'a mut UseTree,
+    }
+
+    let similar_trees = trees.iter_mut().filter_map(|tree| {
+        if tree.share_prefix(&use_tree, merge_by) {
+            let similarity = tree
+                .path
+                .iter()
+                .zip(&use_tree.path)
+                .take_while(|(a, b)| a.equal_except_alias(b))
+                .count();
```

Comment:
I think we should be able to skip the derivation of this value when merge_by is not the SharedPrefix::One variant, correct? If so, then would be good to variant gate this and default to 0 for the others. However, if not then let's add some test cases with the Module and/or Crate variants that demonstrates the necessity.

Reply:
I think that's right! Only `SharedPrefix::One` requires `similarity`. I fixed the code so computing `similarity` will be done in the case of `SharedPrefix::One` only.

> I am reminded of how great it would be to one day be able to rid of all these clones 😞

True... :-|